src/tests/example.spec.js

```diff
@@ -158,4 +158,102 @@ test.describe('Saucedemo app basic tests', () => {
             expect(cartPrice).toBe(selectedProducts[i].price);
         }
     });
+
+    
+    test('check info about several added random products and Total Price on the "Checkout: Overview" page', async (
+        /** @type {{ app: import('../pages/Application').Application }} */{ app },
+    ) => {
+        // get all the elements of the products
+        const products = await app.inventory.product;
+        const productCount = await products.count();
+
+        // generate three unique indexes
+        const randomIndexes = new Set();
+        const uniqueIndexesCount = Math.min(3, productCount);
+        while (randomIndexes.size < uniqueIndexesCount) {
+            const randomIndex = Math.floor(Math.random() * productCount);
+            randomIndexes.add(randomIndex);
+        }
+
+        // store information about selected products
+        const selectedProducts = [];
```

Comment:
Let's move this loop to the page object. Let's avoid using playwright methods (i.e. nth, locator, getByTestId, etc) in the test